### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by Nezerwa**
in this branch i  :
-Added a new JavaScript file and import it as a module.
-Added event listener to the checkbox (change).
-Implemented a function for the "Clear all completed" button (use filter() method).

### CODE REVIEW

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Nezerwa , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [ ] After adding multiple to-dos, when I marked one to-do as complete & refresh the page, the marked to-do got unmarked. Kindly fix it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AshakaE** [comment]
### Hi @Nezerwa  🙋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request ❌ 

You have submitted a project, that has linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by vickymarz — CHANGES_REQUESTED** [review]
Hi @Nezerwa, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project that has failed linter check. KIndly fix your linter errors and submit again 👍🏽 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Eligrand, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Linter checks/ results are not showing in your PR. Please ensure that linter result is green, this means that the last commit have a green check mark (✅) or a summary of the passed checks is in the PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/modules/delete.js [hunk 1]
```diff
@@ -5,15 +5,19 @@ export let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
...
       const item = e.target.parentNode.parentNode;
-      container.removeChild(item);
-      const deletedTask = e.target.previousElementSibling.textContent;
-      tasks = tasks.filter((task) => task.description !== deletedTask);
-      localStorage.setItem('tasks', JSON.stringify(tasks));
+      item.remove();
+      tasks.splice(deletedTaskIndex, 1);
+
+      // update index of remaining tasks
+      tasks.forEach((task, i) => {
```
Comment (christianbiring1) [inline src/modules/delete.js:15]: - [ ] Kidly make sure when the delete icon is pressed, the task is deleted from the DOM and the local storage as well.
`Hint: You may consider say:`
```js
// When the button is clicked you reset the array of tasks by filtering them
//   excluding the one which was deleted and 
 //  then you update the list in your localStorage 
  //  by setting this new array that exclude the deleted task
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @Nezerwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 
soclose


## Highlights
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 
- [ ] Good README :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/delete.js [hunk 2]
```diff
@@ -5,12 +5,16 @@ export let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
...
-    const container = document.querySelector('.container');
     btn.addEventListener('click', (e) => {
+      const deletedTaskIndex = e.target.dataset.index;
       const item = e.target.parentNode.parentNode;
-      container.removeChild(item);
-      const deletedTask = e.target.previousElementSibling.textContent;
-      tasks = tasks.filter((task) => task.description !== deletedTask);
+      item.remove();
+      tasks.splice(deletedTaskIndex, 1);
+
```
Comment (mckent05) [inline src/modules/delete.js:13]: - Just like the previous reviewer suggested, your app doesn't work as expected, first according to the project requirements, when a user clicks on the checkbox it should be updated in the local storage i.e `completed: true`.
trjsrjsrt
- Also, when I click on the `clear completed` button, according to the project requirements, all completed tasks should be removed from the DOM, and all indexes should be updated. Please fix this 👍 👍 

**OPTIONAL** - I noticed that when I delete any tasks, asides from the last task to be added to the DOM, all tasks in the local storage get deleted. Please fix this 🙏 
bfbsfsdfv

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Nezerwa, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉✔🍾**
Hello @Nezerwa 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing the checks.
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**